src/main/preprocessing/preprocessing.py

```diff
@@ -1,103 +1,72 @@
+    return ati_df
 
 
-def handle_ct_and_at(ct_file: str, ct_df: pd.DataFrame, ati_file: str, ati_df: pd.DataFrame,
-                     language: consts.LANGUAGE = consts.LANGUAGE.PYTHON) -> pd.DataFrame:
-    files_from_at = None
-    if ati_df is not None:
-        try:
-            files_from_at = get_files_from_ati(ati_df)
-        except ValueError:
-            ati_df = None
+def is_test_mode(ct_df: pd.DataFrame) -> bool:
+    return ct_df[CODE_TRACKER_COLUMN.TEST_MODE.value].values[0] == TEST_MODE.ON.value
 
-    ct_df[consts.CODE_TRACKER_COLUMN.FILE_NAME.value], does_contain_ct_name \
-        = get_ct_name_from_ati_data(ct_file, language, files_from_at)
-    if ati_df is not None and does_contain_ct_name:
-        ati_id = get_parent_folder_name(ati_file).split('_')[1]
-        ct_df = ath.merge_code_tracker_and_activity_tracker_data(ct_df, ati_df, ati_id)
-        return ct_df
 
-    ati_new_data = pd.DataFrame(ath.get_full_default_columns_for_at(ct_df.shape[0]))
-    ct_df = ct_df.join(ati_new_data)
-    return ct_df
-
-
-def preprocess_data(path: str) -> str:
-    output_directory = get_output_directory(path, consts.PREPROCESSING_OUTPUT_DIRECTORY)
-    folders = get_all_file_system_items(path, data_subdirs_condition, consts.FILE_SYSTEM_ITEM.SUBDIR)
-    for folder in folders:
-        log.info(f'Start handling the folder {folder}')
-        files = get_all_file_system_items(folder, extension_file_condition(consts.EXTENSION.CSV))
-        try:
-            ct_files, ati_file = __separate_ati_and_other_files(files)
-        # Drop the current folder
-        except ValueError:
-            continue
-
-        ati_df = handle_ati_file(ati_file)
-
-        for ct_file in ct_files:
-            ct_df, language = handle_ct_file(ct_file)
-            ct_df = handle_ct_and_at(ct_file, ct_df, ati_file, ati_df, language)
+# The function returns True if the code tracker file was created and False otherwise
+def __handle_ct_files(ct_files: List[str], output_task_path: str) -> bool:
+    max_ct_file = get_file_with_max_size(ct_files)
+    if max_ct_file:
+        new_ct_path = os.path.join(output_task_path, get_name_from_path(max_ct_file))
+        ct_df = pd.read_csv(max_ct_file)
+        if not is_test_mode(ct_df):
+            create_file(get_content_from_file(max_ct_file), new_ct_path)
+            return True
+    return False
 
-            write_result(output_directory, path, ct_file, ct_df)
 
-        log.info(f'Finish handling the folder {folder}')
+# Merge ide-events files
+# Delete unnecessary code tracker files
+# Delete files with test mode = ON
+def preprocessing(path: str) -> str:
+    output_directory = get_output_directory(path, consts.PREPROCESSING_DIRECTORY)
+    user_folders = get_all_file_system_items(path, lambda dir: 'user' in dir, consts.FILE_SYSTEM_ITEM.SUBDIR)
+    for user_folder in user_folders:
+        user_path = os.path.join(path, user_folder)
```

Comment:
user_folder is already a path, why you want to join it? Btw, it works since user_folders starts with the root directory, so ps.path.join doesn't join it with path